ios/Plugin/Plugin.swift

```diff
@@ -8,42 +8,60 @@ import GoogleSignIn
+    private var googleSignInConfiguration: GIDConfiguration!;
+    private var forceAuthCode: Bool = false;
+    private var additionalScopes: [String]!;
     
+    // these are scopes granted by default by the signIn method
+    private let defaultGrantedScopes = ["email", "profile", "openid"];
+
+
+    /// On plugin load, we want to load static config using capacitor.config.js
+    /// In case you don't need to use dynamic configuration, like using different clientIds or scopes in different flows
+    /// then it is a way to configure the plugin. The other way- is the initialize method which will override
+    /// corresponding settings.
     public override func load() {
-        googleSignIn = GIDSignIn.sharedInstance;
-        
-        let serverClientId = getServerClientIdValue();
-        
-        guard let clientId = getClientIdValue() else {
-            NSLog("no client id found in config")
+        guard let staticConfigClientId = self.getClientIdValue() else {
+            NSLog("No client id found in config")
             return;
         }
-
-        googleSignInConfiguration = GIDConfiguration.init(clientID: clientId, serverClientID: serverClientId)
         
-        // these are scopes granted by default by the signIn method
-        let defaultGrantedScopes = ["email", "profile", "openid"];
-
-        // these are scopes we will need to request after sign in
-        additionalScopes = (getConfigValue("scopes") as? [String] ?? []).filter {
-            return !defaultGrantedScopes.contains($0);
-        };
+        googleSignIn = GIDSignIn.sharedInstance;
+        
+        let staticConfigServerClientId = self.getConfigValue("serverClientId") as? String
+        let staticConfigScopes = self.getConfigValue("scopes") as? [String] ?? []
+        
+        self.googleSignInConfiguration = GIDConfiguration.init(clientID: staticConfigClientId,
+                                                          serverClientID: staticConfigServerClientId)
+        self.additionalScopes = self.getAdditionalScopes(scopes: staticConfigScopes)
+        self.forceAuthCode = self.getConfigValue("forceCodeForRefreshToken") as? Bool ?? false
                 
-        if let forceAuthCodeConfig = getConfigValue("forceCodeForRefreshToken") as? Bool {
-            forceAuthCode = forceAuthCodeConfig;
-        }
-
         NotificationCenter.default.addObserver(self, selector: #selector(handleOpenUrl(_ :)), name: Notification.Name(Notification.Name.capacitorOpenURL.rawValue), object: nil);
     }
 
+    /// By passing options in the initialize call, you will override values that were loaded from static config.
+    /// The values that does not overlap will remain unchanged.
+    ///
+    /// This can be used to re-configure the plugin in the runtime.
     @objc
     func initialize(_ call: CAPPluginCall) {
+        let scopes: [String] = call.getArray("scopes", String.self) ?? []
+        let serverClientId: String? = call.getString("serverClientId")
```

Comment:
I think this is constant - i.e. we don't need to pass it through params?

Reply:
I'm thinking about making this change universal. 
This may be constant from the JS point of view (and will be most likely defined this way), but I imagine a case when two flows are using different GCP projects (what was the idea at first I guess?). That would allow doing it so. Or just using different keys for different flows.